lib/health-check/health-check-executor.service.ts

```diff
@@ -0,0 +1,128 @@
+import { Injectable, BeforeApplicationShutdown } from '@nestjs/common';
+import {
+  HealthIndicatorFunction,
+  HealthIndicatorResult,
+} from '../health-indicator';
+import { HealthCheckError } from '../health-check/health-check.error';
+import {
+  HealthCheckResult,
+  HealthCheckStatus,
+} from './health-check-result.interface';
+
+/**
+ * Takes care of the execution of health indicators.
+ *
+ * @description
+ * The HealthCheckExecutor is standalone, so it can be used for
+ * the legacy TerminusBootstrapService and the HealthCheckService.
+ *
+ * On top of that, the HealthCheckExecutor uses the `BeforeApplicationShutdown`
+ * hook, therefore it must implement the `beforeApplicationShutdown`
+ * method as public. We do not want to expose that
+ * to the end-user.
+ *
+ * @internal
+ */
+@Injectable()
+export class HealthCheckExecutor implements BeforeApplicationShutdown {
+  private isShuttingDown = false;
+
+  private async executeHealthIndicators(
+    healthIndicators: HealthIndicatorFunction[],
+  ) {
+    const results: any[] = [];
+    const errors: any[] = [];
+    for (let healthIndicator of healthIndicators) {
+      try {
+        const result = await healthIndicator();
+        result && results.push(result);
+      } catch (error) {
+        // Is not an expected error. Throw further!
+        if (!error.causes) throw error;
+        // Is a expected health check error
+        errors.push((error as HealthCheckError).causes);
+      }
+    }
+
+    return { results, errors };
+  }
+
+  private getSummary(results: any[]): HealthIndicatorResult {
+    return results.reduce(
+      (previous: any, current: any) => Object.assign(previous, current),
+      {},
+    );
+  }
+
+  private getResult(results: any[], errors: any[]): HealthCheckResult {
+    const infoErrorCombined = (results || []).concat(errors || []);
```

Comment:
maybe instead of fallbacks (`results || []`), set the default values in the method signature (or change types to `any[] | undefined`)

Reply:
done!